### DISCUSSION TITLE
Django resful or Golang?

### DISCUSSION THREAD

**Discussion by levinkm** [community/community#35790] [Apps, API and Webhooks]
Which technology would you prefer using for developing a micro-services  system?  Django restful framework or Golang . Please help me understand why and which to choose.

**Comment by ng-druid** [comment]
What has worked well for our organization is serverless golang using serverless framework. We choose golang to begin with for serverless development because of its low memory, cpu footprint desiring to create a highly cost effective enterprise. Our enterprise doesn't pay for running any servers. Instead we host our APIs in the cloud at little to no cost. This model also adheres nicely to sustainable web dev practices like that laid our by the aws well-architected framework. We build all our software with sustainability at center golang has worked very well for us in this regards.

The enterprise actually begin on .NET because that was the knowledge base at the time similar to Django but C#. However, we  quickly realized that running enterprise level .NET APIs has incredibly expensive on our low budget. Seeking something more affordable discovered serverless. Attempted to run .NET on Serverless but was not satisfied with speed. Not sure how much has changed but at the time +2 years ago .NET had a very HIGH start-up time which kills serverless lambda performance. Seeked out a more optimal language, discovered golang rest is history.

Python also isn't a bad choice for running on serverless stack but does consume more energy and resources than golang or rust. Rust is the other serverless language people have been recently talking about. Haven't used it ourself since we have found golang to be more than adequate for our needs.

We will adding carbon awareness to our enterprise very soon and one of the steps will be making our APIs available on every aws data center world wide. This is easily achievable with serverless at very low cost. It would be a completely other story to be running ec2 instances and/or Kubernetes containers on every single data center which Django would require.

If at all intrigued serverless framework is what we use to make development a bit more efficient.

https://www.serverless.com/

This is our enterprise API mono repo

https://github.com/rollthecloudinc/vertigo

We use Bazel to build and also allow for hybrid language lambas. An example of where we have found using node.js over golang is when a specific lib exists in node.js that doesn't exist in golang. In that case using bazel makes it easy to keep everything under one roof using multiple languages. There are also machine learning and AI packages in other languages like Python which we will also be possibly adding into this mix. Bazel is pretty nice for keeping this all nicely housed in one mono repo as independent lambdas.

https://bazel.build/

**Comment by Ralphbaer** [comment]
If you're building an early-stage startup, Django gives you an incredible head start. You get batteries-included productivity: an opinionated framework, ORM, authentication, and an entire ecosystem that lets you move fast. If your competitive edge isn’t in raw performance but in product, user experience, or go-to-market strategy, Django lets you iterate rapidly and ship quickly.

Now, if your company is scaling and performance, efficiency, or concurrency become key constraints, Go is an excellent alternative. It’s designed for speed, efficiency, and developer productivity at scale. Some of the most important infrastructure companies—Kubernetes, CockroachDB, HashiCorp—are built in Go because it lets teams ship fast while maintaining system simplicity.
